android/expoview/src/main/java/host/exp/exponent/experience/ErrorActivity.kt

```diff
@@ -0,0 +1,164 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.experience
+
+import android.content.Context
+import androidx.viewpager.widget.ViewPager
+import javax.inject.Inject
+import android.os.Bundle
+import host.exp.exponent.di.NativeModuleDepsProvider
+import android.content.Intent
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+import host.exp.exponent.Constants
+import host.exp.exponent.LauncherActivity
+import host.exp.exponent.analytics.Analytics
+import host.exp.exponent.kernel.ExponentError
+import host.exp.exponent.kernel.Kernel
+import host.exp.expoview.databinding.ErrorActivityNewBinding
+import java.util.*
+
+class ErrorActivity() : FragmentActivity() {
+  private lateinit var binding: ErrorActivityNewBinding
+
+  private lateinit var pager: ViewPager
+
+  private var manifestUrl: String? = null
+
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var kernel: Kernel
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    binding = ErrorActivityNewBinding.inflate(layoutInflater)
+    val view = binding.root
+    setContentView(view)
+    pager = binding.errorViewPager
+
+    NativeModuleDepsProvider.getInstance().inject(ErrorActivity::class.java, this)
+
+    ExperienceActivity.removeNotification(this)
+
+    val bundle = intent.extras
+    manifestUrl = bundle!!.getString(MANIFEST_URL_KEY)
+    if (manifestUrl == null && Constants.INITIAL_URL != null) {
+      manifestUrl = Constants.INITIAL_URL
+    }
```

Comment:
manifestUrl can be null if Constants.INITIAL_URL is null.

```suggestion
    manifestUrl = bundle!!.getString(MANIFEST_URL_KEY) ?: Constants.INITIAL_URL
```